app/src/main/java/com/example/androidtraining/ui/product/ProductScreen.kt

```diff
@@ -0,0 +1,237 @@
+                        Icon(
+                            imageVector = Icons.Outlined.ArrowBack,
+                            contentDescription = stringResource(id = R.string.arrow_back)
+                        )
+                    }
+                },
+                actions = {
+                    IconButton(onClick = { /* doSomething() */ }) {
+                        BadgedBox(
+                            badge = {
+                                Badge(
+                                    contentColor = White,
+                                    containerColor = Purple,
+                                ) {
+                                    val badgeNumber = "8"
+                                    Text(
+                                        badgeNumber,
+                                        color = White
+                                    )
+                                }
+                            }) {
+                            Icon(
+                                Icons.Outlined.ShoppingCart,
+                                contentDescription = stringResource(id = R.string.shopping_cart)
+                            )
+                        }
+                    }
+                }
+            )
+        },
+        content = { paddingValues ->
+            viewModel.productResponse.value?.getOrNull()?.let { item ->
+                Column(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .paint(
+                            painterResource(id = R.drawable.product_background),
+                            contentScale = ContentScale.FillWidth
+                        )
+                        .padding(horizontal = paddingLarge)
+                        .padding(top = paddingValues.calculateTopPadding())
+                ) {
+                    Box(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .clip(RoundedCornerShape(16.dp))
+                    ) {
+                        Row(
+                            modifier = Modifier
+                                .fillMaxWidth()
+                        ) {
+                            Image(
+                                modifier = Modifier,
+                                painter = rememberAsyncImagePainter(item.image),
+                                contentDescription = item.title
+                            )
+                        }
+
+                        Row(
+                            modifier = Modifier
+                                .padding(paddingMedium)
+                                .fillMaxWidth(),
+                            horizontalArrangement = Arrangement.End
+                        ) {
+                            Box(
+                                modifier = Modifier
+                                    .height(30.dp)
```

Comment:
Look for any `.dp` in the code usages and refactor them as you have others. Try to reuse as much as you can.